packages/nuxt/src/app/nuxt.ts

```diff
@@ -45,6 +45,7 @@ export interface RuntimeNuxtHooks {
   'page:finish': (Component?: VNode) => HookResult
   'page:transition:start': () => HookResult
   'page:transition:finish': (Component?: VNode) => HookResult
+  'page:view-transition:start': (transition: ViewTransition) => HookResult
```

Comment:
How about providing an exported type in nuxt and importing it here?
```ts
export interface ViewTransition {
    ready: Promise<void>
    finished: Promise<void>
    updateCallbackDone: Promise<void>
  }
  ```
  
  ```ts
import {ViewTransition} from "wherever you'll place it"
```

Reply:
Yeah I think I overthinked this one 😅 I tought it has to be importable as "import type {ViewTransition} from 'nuxt'"

Reply:
Happens to the best of us =)